### ISSUE TITLE
XMLSchema.validate() only works with lxml - TypeError otherwise

### ISSUE THREAD

**Issue by gsauthof** [brunato/xmlschema#80]
With the current xmlschema-1.0.3 I ran into this issue (while trying out some of the API that is described in the documentation: https://xmlschema.readthedocs.io/en/latest/usage.htm):

Validating an invalid XML file works as expected:

    In [1]: import xmlschema
    In [2]: xmlschema.validate('mini-invalid.xml', 'mini.xsd')
    XMLSchemaValidationError: failed validating '1.23' with XsdPatternsFacet(['[0-9]+']):

But doing the same for an ElmentTree tree fails:

    In [3]: import xml.etree.ElementTree as ET
    In [4]: t = ET.parse('mini-invalid.xml')
    In [5]: xmlschema.validate(t, 'mini.xsd')
    <class 'str'>: (<class 'TypeError'>, TypeError("Type 'Element' cannot be serialized.",))

It does work as expected when using LXML, though:

    In [6]: import lxml
    In [7]: u = lxml.etree.parse('mini-invalid.xml')
    In [8]: xmlschema.validate(u, 'mini.xsd')
    XMLSchemaValidationError: failed validating '1.23' with XsdPatternsFacet(['[0-9]+']):

In the ET case, the information seems to be all there - it just fails during  generating the string:

    In [18]: xsd = xmlschema.XMLSchema('mini.xsd')
    In [19]: list(xsd.iter_errors(t))
    Out[19]: 
    [xmlschema.validators.exceptions.XMLSchemaValidationError(XsdPatternsFacet(['[0-9]+']),
                                                          '1.23')]

I can provide you the mini.xsd and min-invalid.xml files (they are small) but I think that you can easily reproduce this issue with any XSD in combination with some non-validating XML file.

**Comment by brunato** [comment]
It seems a problem with `ElementTree` importing that doesn't import the accelerated C API at the end when you perform `import xml.etree.ElementTree` after `import xmlschema`. So the library founds a different Element class and raises an error when try to convert the error to string.

This reproduces your test:

``` python
>>> import xmlschema
>>> import xml.etree.ElementTree as ET
>>> t = ET.parse('vehicles-2_errors.xml')
>>> xmlschema.validate(t, 'vehicles.xsd')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/brunato/Development/projects/xmlschema/xmlschema/documents.py", line 44, in validate
    schema.validate(xml_document, use_defaults, namespaces)
  File "/home/brunato/Development/projects/xmlschema/xmlschema/validators/xsdbase.py", line 497, in validate
    raise error
xmlschema.validators.exceptions.XMLSchemaValidationError: <exception str() failed>
```

but if I change the import order the result is correct:

```python
>>> import xml.etree.ElementTree as ET
>>> import xmlschema
>>> t = ET.parse('vehicles-2_errors.xml')
>>> xmlschema.validate(t, 'vehicles.xsd')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/brunato/Development/projects/xmlschema/xmlschema/documents.py", line 44, in validate
    schema.validate(xml_document, use_defaults, namespaces)
  File "/home/brunato/Development/projects/xmlschema/xmlschema/validators/xsdbase.py", line 497, in validate
    raise error
xmlschema.validators.exceptions.XMLSchemaValidationError: failed validating <Element '{http://example.com/vehicles}cars' at 0x7f63e6a8fae8> with XsdGroup(model='sequence', occurs=[1, 1]):

Reason: character data between child elements not allowed!

Schema:

  <xs:complexType xmlns:xs="http://www.w3.org/2001/XMLSchema">
    <xs:sequence>
      <xs:element maxOccurs="unbounded" minOccurs="0" name="car" type="vh:vehicleType" />
    </xs:sequence>
  </xs:complexType>

Instance:

  <cars xmlns="http://example.com/vehicles">
    CHARACTER DATA
    <car make="Porsche" model="911" />
    <car make="Porsche" model="911" />
  </cars>

Path: /{http://example.com/vehicles}vehicles/{http://example.com/vehicles}cars

>>> 
```

You can reproduce this also checking the ET.Element class vs ET._Element_Py (the real Python Element class in Python 3).

If the import of ElementTree is after xmlschema the accelerated Element is not imported:
```python
>>> import xmlschema
>>> import xml.etree.ElementTree as ET
>>> ET.Element is ET._Element_Py
True
```
otherwise the C accelerated Element is imported:

```python
>>> import xml.etree.ElementTree as ET
>>> import xmlschema
>>> ET.Element is ET._Element_Py
False
```

I'm trying to understand if there is a workaround for this.

Thanks,
Davide

**Comment by brunato** [comment]
Should related to the import of the _defusedxml.ElementTree_ library performed by the _etree.py_ module.
Infact if the _defusedxml.ElementTree_ library is imported before _xml.etree.ElementTree_ the C optimized version is not loaded:

```python
>>> import defusedxml.ElementTree
>>> import xml.etree.ElementTree as ET
>>> ET.Element is ET._Element_Py
True
```
Viceversa works normally:

```python
>>> import xml.etree.ElementTree as ET
>>> import defusedxml.ElementTree
>>> ET.Element is ET._Element_Py
False
```
I've opened an [issue](https://github.com/tiran/defusedxml/issues/28) for this.

Best,
Davide

**Comment by brunato** [comment]
As responded by the author of *defusedxml*, the impact of this problem cannot be avoided, because the pure Python ElementTree is needed by this library. Also, as the author wrote, the project is pretty much abandoned because the functionalities will be included in the next versions of Python 2.7, 3.6 and 3.7.

For *xmlschema* I've decided to mitigate the impact of this using a lazy import of defusedxml.ElementTree, wrapping the library in a class that take care of this.

**Comment by brunato** [comment]
The problem should be mitigated with a lazy import *defusedxml* (class `LazyDefusedElementTree` in module *etree.py*). Also the pure python *ElementTree* module is loaded after *defusedxml* in order to use it for cases like this for etree elements string serialization.
Thanks
